new project with Sphinx theme

lauranovich opened the pull request:
new project 
add sphinx theme and a basic page

lauranovich committed:
new project with Sphinx theme

tzach commented:
Nice start!

tzach commented:
Next tasks:
- Deploy to https://scylladb.github.io/scylla-monitoring 
- Copy latest doc to the master branch

Once this is working, we can add more releases.

lauranovich commented:
What elements do we want on the header?
how does it work with links?

tzach commented:
> What elements do we want on the header?

See example here:
https://scylladb.github.io/python-driver/master/index.html
(replace https://scylladb.github.io/python-driver/ with https://scylladb.github.io/scylla-monitoring)

> how does it work with links?

Not sure I understand the question. 
See example here: https://github.com/scylladb/python-driver/blob/master/docs/conf.py#L103

lauranovich committed:
new content for 3.4

lauranovich committed:
added sphinx build to git ignore

lauranovich committed:
changed header links

tzach commented on .gitignore:
```diff
@@ -0,0 +1 @@
+/docs/_build/
```
Please remove everything under _build and add the following to .gitignore:
docs/_build/
docs/venv/
*.swp
*.pyc
docs/bin/
docs/include/
docs/lib/
docs/lib64
docs/pip-selfcheck.json
docs/.Python

lauranovich committed:
moved files out of version folder, adjusted index, fixed config

lauranovich committed:
changed config and make file to eliminate errors

lauranovich committed:
moved files to eliminate errors

tzach commented on pyproject.toml:
```diff
@@ -0,0 +1,22 @@
+[tool.poetry]
+name = "sphinx-docs"
```
```suggestion
name = "scylla-monitoring-docs"
```

tzach commented on pyproject.toml:
```diff
@@ -0,0 +1,22 @@
+[tool.poetry]
+name = "sphinx-docs"
+description = "ScyllaDB Documentation"
```
```suggestion
description = "Scylla Monitoring Documentation"
```

tzach commented on pyproject.toml:
```diff
@@ -0,0 +1,22 @@
+[tool.poetry]
+name = "sphinx-docs"
+description = "ScyllaDB Documentation"
+version = "1.3"
```
```suggestion
version = "0.1"
```

tzach commented on pyproject.toml:
```diff
@@ -0,0 +1,22 @@
+pyyaml = "^5.3"
+pygments = "2.2.0"
+recommonmark = "0.5.0"
+sphinx-scylladb-theme = "0.1.6"
```
```suggestion
sphinx-scylladb-theme = "0.1.8"
```

tzach commented:
File `docs/_utils/__pycache__/cql.cpython-38.pyc` and all pyc files should not be part of Git

tzach commented on docs/_utils/deploy.sh:
```diff
@@ -0,0 +1,17 @@
+#!/bin/sh -e
+
+ORIGIN="https://github.com/scylladb/scylla-docs.git"
```
```suggestion
ORIGIN="https://github.com/scylladb/scylla-monitoring.git"
```

tzach commented on docs/_utils/redirections.yaml:
```diff
@@ -0,0 +1,258 @@
+### a dictionary of redirections
```
This file is not relevant for this project

lauranovich replied:
ok I'll remove it

tzach commented on docs/index.rst:
```diff
@@ -0,0 +1,37 @@
+* `Troubleshoot Scylla Monitor <monitor_troubleshoot>`_
+* :doc:`CQL Optimization Dashboard <cql_optimization>`
+* :doc:`Adding and Modifying Dashboards <updating_dashboard>`
+* :doc:`Alerting <alerting>`
```
The original page also include:
```
**Additional Information**:

* `Scylla Monitoring Support Matrix <matrix>`_
* `Upgrade Scylla Monitoring </upgrade/upgrade-monitor/>`_
* `Monitoring lesson <https://university.scylladb.com/courses/scylla-operations/lessons/admin-procedures-and-basic-monitoring/>`_ on Scylla University

.. image:: monitor.png
```

tzach commented on docs/index.rst:
```diff
@@ -0,0 +1,37 @@
+* `prometheus` - collects and stores metrics
+* `alertmanager` - handles alerts
+* `grafana` - dashboard server
+
```
For older version of Scylla Monitoring see `here <https://docs.scylladb.com/operating-scylla/monitoring/>`_

tzach commented on docs/monitor_troubleshoot.rst:
```diff
@@ -0,0 +1,209 @@
+The preferred way of running the container is using a non root user.
+See the `monitoring`_ Docker post-installation section.
+
+.. _`monitoring`: /operating-scylla/monitoring/3.4/monitoring_stack/#docker-post-installation 
```
```suggestion
.. _`monitoring`: /monitoring_stack/#docker-post-installation 
```

Need to check
All link should be update to this project and not include version

tzach commented on docs/monitor_without_docker.rst:
```diff
@@ -0,0 +1,379 @@
+
+The following procedure uses a ``CentOS 7`` based instance
+
+1. Download the latest Scylla Monitoring release. Currently it is |mon_version|.
```
```suggestion
1. Download the latest Scylla Monitoring release.
```

Remove all version info and |mon_version| version

lauranovich committed:
Update pyproject.toml

Co-authored-by: Tzach Livyatan <tzach.livyatan@gmail.com>

lauranovich committed:
Update pyproject.toml

Co-authored-by: Tzach Livyatan <tzach.livyatan@gmail.com>

lauranovich committed:
Update pyproject.toml

Co-authored-by: Tzach Livyatan <tzach.livyatan@gmail.com>

lauranovich committed:
Update pyproject.toml

Co-authored-by: Tzach Livyatan <tzach.livyatan@gmail.com>

lauranovich committed:
Update docs/_utils/deploy.sh

Co-authored-by: Tzach Livyatan <tzach.livyatan@gmail.com>

lauranovich committed:
Update docs/monitor_troubleshoot.rst

Co-authored-by: Tzach Livyatan <tzach.livyatan@gmail.com>

lauranovich committed:
Update docs/monitor_without_docker.rst

Co-authored-by: Tzach Livyatan <tzach.livyatan@gmail.com>

lauranovich committed:
added links to other topics

lauranovich committed:
inserted text in PR via comments and now need to resyc
Merge branch 'add-sphinx' of https://github.com/lauranovich/scylla-monitoring into add-sphinx

lauranovich commented:
@tzach - Hope I didn't mess this one up. Please review

lauranovich committed:
removed contents of redirections file to prevent errors

lauranovich committed:
removed version tag

tzach commented:
I'm getting the following error:
```
Because scylla-monitoring-docs depends on sphinx-scylladb-theme (0.1.8) which doesn't match any versions, version solving failed.
```

But there is a 0.1.8 version https://pypi.org/project/sphinx-scylladb-theme/

@dgarcia360 any idea?

lauranovich commented:
@tzach I have the same errror message
@dgarcia360 - please let me know what to do

dgarcia360 commented:
@tzach @lauranovich

Before the error, you will see the following warning ``The lock file is not up to date with the latest changes in pyproject.toml.`` This appears when the ``pyproject.yml`` file has been edited manually but the ``poetry.lock`` file has not been updated accordingly.

To solve the error, you can either:

a) (Recommended) Run ``poetry update`` in the root project folder.
b) Delete the file ``poetry.lock`` and run ``make preview`` again.

**Extra info about how poetry works**

I'd like to explain to you why both files should match, it could help to understand better how poetry works and solve other problems in the future. As you have seen, Poetry works mainly with two files:

-  ``pyproject.yml``:  Describes the minimum/maximum/exact version of each dependency to be installed. Example: sphinx-scylladb-theme<0.1.6.
- ``poetry.lock``: Describes the exact version being used to avoid installing modules from the same ``pyproject.yml`` result in two different installs. This is auto-generated from ``pyproject.yml``. Example: sphinx-scylladb-theme==0.1.5, which is < 0.1.6.

The current [pyproject.yml](https://github.com/scylladb/scylla-monitoring/pull/984/files#diff-522adf759addbd3b193c74ca85243f7dR12) file currently points to sphinx-scylladb-theme==**0.1.8** but the dependency in [poetry.lock](https://github.com/scylladb/scylla-monitoring/pull/984/files#diff-41fe8bebc1a2a52eb5321b759e40b3a8R748) still is **0.1.6**, which is different to 0.1.8. When we run ``poetry update``, the file ``poetry.lock`` will be regenerated according to the latest ``pyproject.yml`` specification.

lauranovich committed:
removed redirections

tzach commented:
@amnonh this looks good to me.
I think we can merge, there is no, or very little risk

lauranovich commented:
I am correcting an error on my end, I will need to send a new commit - wait before you merge please

lauranovich committed:
modify git ignore file

lauranovich commented:
@tzach @amnonh  - all set - can we merge?

amnonh commented:
I've tested it locally and saw that it works for me
